File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/Visitor/Title.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Visitor {
    internal class Title : IVisited {
        private readonly string text;

        public void AcceptVisitor(IVisitor v) {
            v.VisitTitleStart(this);
            v.VisitTitleEnd();
        }

        public Title(String text) {
            this.text = text;
        }

        public override string ToString() {
            return text;
        }
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/visitor/Title.java
package patterns.visitor;

public class Title implements Visited {
	private String text;
	public void acceptVisitor(Visitor v) {
		v.visitTitleStart(this);
		v.visitTitleEnd();
	}
	public Title(String text) {
		this.text = text;
	}
	public String toString() {
		return text;
	}
}
